### PR TITLE
JUnit 5 Dynamic tests not showing with their displayName

### PR DESCRIPTION
**Pull request by robmoore-i**
### Context

Resurrection of https://github.com/gradle/gradle/pull/21361 made by @uberto, which is for https://github.com/gradle/gradle/issues/5975
- The fix is from @uberto
- Applied the suggestions from @6hundreds for the test
- I made a [previous PR](https://github.com/gradle/gradle/pull/23170), but since this is a bug fix that people care about, I've retargeted it for 7.x (which involved closing the previous PR, which targeted master)

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

(Checklist notes: No docs changes necessary. The added test in `JUnitPlatformLoggingIntegrationTest` proves that there is a bug, hence no unit test changes other than updating `AbstractTestLoggerTest`.)

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation

### CODE REVIEW

**Comment by robmoore-i** [comment]
@bot-gradle test RFM

**Comment by robmoore-i** [comment]
@6hundreds Tagging you here since you reviewed the PR from which this is resurrected. I applied your suggestions, which all looked sound to me. PTAL

**Comment by 6hundreds** [comment]
@robmoore-i Hi! LGTM.
Let's ask @gradle/bt-jvm to also take a quick look on it

**Review by 6hundreds — APPROVED** [review]

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformLoggingIntegrationTest.groovy [hunk 1]
```diff
@@ -98,4 +100,33 @@ class JUnitPlatformLoggingIntegrationTest extends JUnitPlatformIntegrationSpec
...
+                        .mapToObj(v -> dynamicTest(v + " is even", () -> assertEquals(0, v % 2)));
+                }
+            }
+        """
+
+        when:
+        runAndFail("test")
+
+        then:
+        def prefix = "JUnitJupiterDynamicTest > streamOfTests() > org.gradle.JUnitJupiterDynamicTest"
```
Comment (marcphilipp) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformLoggingIntegrationTest.groovy:127]: ❓ Should this really include the test class name?

```suggestion
        def prefix = "JUnitJupiterDynamicTest > streamOfTests() > "
```

Comment (robmoore-i) [reply subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformLoggingIntegrationTest.groovy:127]: You're right, it shouldn't. I'll look into why the display name is including the class name in this case.

**Review by tresat — APPROVED** [review]

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformLoggingIntegrationTest.groovy (hunk 1 above)
Comment (robmoore-i) [reply subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformLoggingIntegrationTest.groovy:127]: Fixed 👍

**Comment by robmoore-i** [comment]
Closing to change the branch in the fork, to prevent confusion.

**Pull request closed without merging**